frontend/packages/dev-console/src/components/import/git/GitSection.tsx

```diff
@@ -114,95 +171,71 @@ const GitSection: React.FC<GitSectionProps> = ({
         values.application.selectedKey !== UNASSIGNED_KEY &&
         setFieldValue('application.name', `${gitRepoName}-app`);
 
-      if (buildStrategy === BuildStrategyType.Devfile && !values.devfile?.devfileSourceUrl) {
-        // No need to check the existence of the file, waste of a call to the gitService for this need
-        const devfileContents = gitService && (await gitService.getDevfileContent());
-        if (!devfileContents) {
-          setFieldValue('devfile.devfileContent', null);
-          setFieldValue('devfile.devfileHasError', true);
-          setValidated(ValidatedOptions.error);
-        } else {
-          setFieldValue('devfile.devfileContent', devfileContents);
-          setFieldValue('devfile.devfileHasError', false);
-          setValidated(ValidatedOptions.success);
-        }
+      setFieldValue('import.loaded', loaded);
+      setFieldValue('import.loadError', loadError);
+      setFieldValue('import.strategies', importStrategies);
+      if (importStrategies.length > 0) {
+        values.formType !== 'edit' && setFieldValue('import.showEditImportStrategy', false);
+        setFieldValue('import.selectedStrategy', importStrategies[0]);
+        setFieldValue('import.recommendedStrategy', importStrategies[0]);
       } else {
-        setValidated(ValidatedOptions.success);
+        setFieldValue('import.selectedStrategy', {
+          name: 'Devfile',
+          type: ImportStrategy.DEVFILE,
+          priority: 2,
+          detectedFiles: [],
+        });
+        setFieldValue('import.recommendedStrategy', null);
+        values.formType !== 'edit' && setFieldValue('import.showEditImportStrategy', true);
       }
 
+      if (importStrategies.length > 0) {
+        switch (importStrategies[0].type) {
```

Comment:
sorry, i missed why it is [0]?

Reply:
Because the first strategy will be of the highest priority